forte/data/readers/classification_reader.py

```diff
@@ -0,0 +1,177 @@
+from forte.common.configuration import Config
+from forte.data.data_pack import DataPack
+from forte.data.base_reader import PackReader
+from ft.onto.base_ontology import *
+
+
+__all__ = ["ClassificationDatasetReader"]
+
+
+class ClassificationDatasetReader(PackReader):
+    r"""
+    A generic dataset reader class for classification datasets.
+
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.index2class = None
+        self.label2index = None
+
+    def set_up(self):
+        assert self.configs.index2class is not None, (
+            "User must set index2class to enable"
+            " the dataset reader encode labels correctly"
+        )
+
+        # class and index
+        if not self.configs.digit_label:
+            # initialize class
+            self.index2class = self.configs.index2class
+            self.label2index = {
+                v: k for k, v in self.index2class.items()
+            }
+
+    def initialize(self, resources: Resources, configs: Config):
+        super().initialize(resources, configs)
+        self.set_up()
+
+    def _collect(self, csv_file: str) -> Iterator[Tuple[int, str]]:
+        with open(csv_file, encoding="utf-8") as f:
+            data = csv.reader(f, delimiter=",", quoting=csv.QUOTE_ALL)
+            next(data)
+            for line_id, line in enumerate(data):
+                yield line_id, line
+
+    def _cache_key_function(self, line_info: Tuple[int, str]) -> str:
+        return str(line_info[0])
+
+    def _parse_pack(self, line_info: Tuple[int, str]) -> Iterator[DataPack]:
+        line_id, line = line_info
+        # content label title
+        pack = DataPack()
+
+        # subtext fields must follow the ontology names
+        data_fields = self.configs.data_fields
+        assert (
+            "label" in data_fields
+        ), "There must be label data field in reader config."
+        assert len(self.configs.data_fields) == len(line), (
+            "Data fields provided in config "
+            "is not aligned with the actual line info from dataset."
+            + str((data_fields, line))
+        )
+        assert (
+            len(self.configs.subtext_fields) > 0
+        ), "There must be at least one subtext to reader to select from"
+
+        df_dict = OrderedDict()
+        for df, value in zip(data_fields, line):
+            df_dict[
+                df
+            ] = value  # in general, value can be subtext or other types of data
+
+        # it determines the order of concatenation
+        subtext_fields = self.configs.subtext_fields
+        # get text and subtext indices
+        assert set(self.configs.subtext_fields).issubset(
+            set(self.configs.data_fields)
+        ), "subtext fields must be a subset of data fields"
+
+        text, subtext_indices = generate_text_n_subtext_indices(
+            subtext_fields, df_dict
+        )
+        pack.set_text(text, replace_func=self.text_replace_operation)
```

Comment:
use `self.set_text` instead of this, so you don't need to call `text_replace_operation` yourself.